precompiles/assets-erc20/src/lib.rs

```diff
@@ -0,0 +1,444 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is 	part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(test, feature(assert_matches))]
+
+use evm::{executor::PrecompileOutput, Context, ExitError, ExitSucceed};
+use frame_support::traits::fungibles::Inspect;
+use frame_support::traits::OriginTrait;
+use frame_support::{
+	dispatch::{Dispatchable, GetDispatchInfo, PostDispatchInfo},
+	sp_runtime::traits::StaticLookup,
+};
+
+use pallet_evm::{AddressMapping, Precompile};
+use precompile_utils::{
+	error, Address, EvmDataReader, EvmDataWriter, EvmResult, Gasometer, LogsBuilder, RuntimeHelper,
+};
+
+use slices::u8_slice;
+use sp_core::{H160, U256};
+use sp_std::{
+	convert::{TryFrom, TryInto},
+	marker::PhantomData,
+	vec,
+};
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+/// Solidity selector of the Transfer log, which is the Keccak of the Log signature.
+pub const SELECTOR_LOG_TRANSFER: &[u8; 32] =
+	u8_slice!("0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef");
+
+/// Solidity selector of the Approval log, which is the Keccak of the Log signature.
+pub const SELECTOR_LOG_APPROVAL: &[u8; 32] =
+	u8_slice!("0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925");
+
+/// Alias for the Balance type for the provided Runtime and Instance.
+pub type BalanceOf<Runtime, Instance = ()> = <Runtime as pallet_assets::Config<Instance>>::Balance;
```

Comment:
```suggestion
pub type BalanceOf<Runtime> = <Runtime as pallet_assets::Config>::Balance;
```

I don't think we need to mention the `Instance` part here since we only have one instance in our runtime.

Reply:
Ohh, maybe the point is that the preocmpiles should be general enough that they can be used in other runtimes where there are multiple instances. I support that.